### PR TITLE
Improve ui::Slider

### PR DESCRIPTION
**Pull request by zilongshanren**
1. Add two APIs to gain big precision
2. Add three type events in Slider's callback.

Fixed issue 
 https://github.com/cocos2d/cocos2d-x/issues/12350
https://github.com/cocos2d/cocos2d-x/issues/12351

### CODE REVIEW

**Comment by zilongshanren** [comment]
@VisualSJ  @pandamicro  please review this PR. I think Web should also support it.

File: cocos/ui/UISlider.h [hunk 1]
```diff
@@ -61,7 +61,11 @@ class CC_GUI_DLL Slider : public Widget
      */
     enum class EventType
     {
-        ON_PERCENTAGE_CHANGED
+        ON_PERCENTAGE_CHANGED,
+        //@since v3.7
+        ON_PERCENTAGE_DOWN,
+        ON_PERCENTAGE_UP,
+        ON_PERCENTAGE_CANCEL
```
Comment (pandamicro) [inline cocos/ui/UISlider.h]: Percentage down/up/cancel is meaningless
I suggest
ON_SLIDEBALL_DOWN
ON_SLIDEBALL_UP
ON_SLIDEBALL_CANCEL

**Comment by pandamicro** [comment]
It's indeed a useful callback, I agree it should be merged to web

File: cocos/ui/UISlider.h (hunk 1 above)
Comment (zilongshanren) [inline cocos/ui/UISlider.h]: Yeah, I agree. I will change them. Thanks.

**Comment by pandamicro** [comment]
@VisualSJ Please sync the API and modifications to web engine

File: cocos/ui/UISlider.cpp [hunk 2]
```diff
@@ -384,18 +385,28 @@ void Slider::loadSlidBallTextureDisabled(SpriteFrame* spriteframe)
...
-        percent = 100;
+        percent = _maxPercent;
     }
     if (percent < 0)
     {
         percent = 0;
     }
     _percent = percent;
-    float res = percent / 100.0f;
+    float res = 1.0 * percent / _maxPercent;
```
Comment (VisualSJ) [inline cocos/ui/UISlider.cpp:409]: set up 50%, Should be 50% of the total length. Not 50% of the maximum length.

**Pull request merged**